kube-client/src/lib.rs

```diff
@@ -611,4 +611,141 @@ mod test {
         csr.delete(csr_name, &DeleteParams::default()).await?;
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore = "needs cluster for ephemeral containers operations"]
+    async fn can_operate_on_ephemeral_containers() -> Result<(), Box<dyn std::error::Error>> {
+        let pod: Pod = serde_json::from_value(serde_json::json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "ephemeral-container-test",
+                "labels": { "app": "kube-rs-test" },
+            },
+            "spec": {
+                "restartPolicy": "Never",
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 2"],
+                }],
+            }
+        }))?;
+
+        let pod_name = pod.name_any();
+        let client = Client::try_default().await?;
+        let pods = Api::<Pod>::default_namespaced(client);
+
+        // Ephemeral containes can only be applied to a running pod, so one must
+        // be created before any operations are tested.
+        match pods.create(&Default::default(), &pod).await {
+            Ok(o) => assert_eq!(pod.name_unchecked(), o.name_unchecked()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        let current_ephemeral_containers = pods
+            .get_ephemeral_containers(&pod.name_any())
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers;
+
+        // We expect no ephemeral containers initially, get_ephemeral_containers should
+        // reflect that.
+        assert_eq!(current_ephemeral_containers, None);
```

Comment:
hm, maybe. the ephemeral container exits pretty quickly right? if that's the case it will likely fix itself for users running integration tests locally - and it doesn't really matter on CI.

Reply:
They do exit quickly, but the Pod is not cleaned up. It will just stay as completed, requiring manual cleanup.

I could just add a "delete pod if it already exists" to the start of the test. :thinking:

Reply:
I see this is done and documented well now :+1: